src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,161 @@
+import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
+import uk.gov.hmcts.ccd.data.casedetails.search.SortOrderField;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition;
+import uk.gov.hmcts.ccd.domain.service.aggregated.AuthorisedGetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.GetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.SearchQueryOperation;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Service
+@Slf4j
+public class ElasticsearchQueryHelper {
+
+    private static final String SORT = "sort";
+    private static final String CASE_DATA_PREFIX = "data.";
+    private static final String COLLECTION_VALUE_SUFFIX = ".value";
+    private static final String KEYWORD_SUFFIX = ".keyword";
+
+    private final ObjectMapper objectMapper;
+    private final ApplicationParams applicationParams;
+    private final ObjectMapperService objectMapperService;
+    private final SearchQueryOperation searchQueryOperation;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final ElasticsearchMappings elasticsearchMappings;
+
+    @Autowired
+    public ElasticsearchQueryHelper(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
+                                    ApplicationParams applicationParams,
+                                    ObjectMapperService objectMapperService,
+                                    SearchQueryOperation searchQueryOperation,
+                                    @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) GetCaseTypeOperation getCaseTypeOperation,
+                                    ElasticsearchMappings elasticsearchMappings) {
+        this.objectMapper = objectMapper;
+        this.applicationParams = applicationParams;
+        this.objectMapperService = objectMapperService;
+        this.searchQueryOperation = searchQueryOperation;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.elasticsearchMappings = elasticsearchMappings;
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(List<String> caseTypeIds, String jsonSearchRequest) {
+        if (CollectionUtils.isEmpty(caseTypeIds)) {
+            throw new BadSearchRequest("At least one case type ID is required.");
+        }
+
+        JsonNode searchRequest = validateAndConvertRequest(jsonSearchRequest);
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(caseTypeIds)
+            .withSearchRequest(searchRequest)
+            .build();
```

Comment:
minor: if we moved this code inside the `CaseSearchOperation.execute` we could have the external search API be completely responsible to create the CrossCaseTypeSearchRequest. We just delegate to the external. What do you think? too much stuff to move around?

Reply:
Did look into this potential but it did require more than expected so I have just put back to the controllers.